### PR TITLE
Use spotless for CI lint

### PR DESCRIPTION
**Pull request by yasith**
Added Lint check to CI using Spotless

Seprated Build/Test/Lint into 3 separate jobs, that can run in parallel.

image

### CODE REVIEW

**Comment by yasith** [comment]
Current dependencies on/for this PR:
* main
  * **PR #52** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/52" target="_blank">Graphite</a> 
    * **PR #51** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/51" target="_blank">Graphite</a> 
      * **PR #53** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/53" target="_blank">Graphite</a> 
        * **PR #54** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/54" target="_blank">Graphite</a> 
          * **PR #56** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/56" target="_blank">Graphite</a> 
            * **PR #57** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/57" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/google/jetpack-camera-app/57?utm_source=stack-comment).

File: .github/workflows/BuildAndTest.yaml [hunk 1]
```diff
@@ -1,50 +0,0 @@
-name: Build and Test
-
-on: [push, pull_request]
```
Comment (yasith) [inline .github/workflows/BuildAndTest.yaml:3]: This was causing the jobs to run twice per each PR update, so I changed it to just push

File: .github/workflows/BuildAndTest.yaml [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
-        uses: gradle/gradle-build-action@v2
-
-      - name: Check lint
-        run: ./gradlew lint
-
-      - name: Build all build type and flavor permutations
-        run: ./gradlew assemble
-
-      - name: Run local tests
-        run: ./gradlew test
```
Comment (yasith) [inline .github/workflows/BuildAndTest.yaml:36]: Split this into a separate job, so it runs in parallel.

Also makes it clearer in the interface about what checks are passing/failing/in-progress.

Screenshot 2023-10-12 at 12 33 55 AM

File: .github/workflows/PushWorkflow.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      - name: Build all build type and flavor permutations
+        run: ./gradlew assemble --parallel --build-cache
+
+      - name: Upload build outputs (APKs)
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-outputs
+          path: app/build/outputs
+
+      - name: Upload build reports
```
Comment (temcguir) [inline .github/workflows/PushWorkflow.yaml:44]: With this change, the "Unit Tests" task will no longer uploads test reports (stored in `build/reports/tests`) that are generated on failure. We should add an "Upload unit test reports" using `if: failure()` in the `tests` job.

File: .github/workflows/PushWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        if: always()
+        continue-on-error: true
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-reports
+          path: app/build/reports
```
Comment (temcguir) [inline .github/workflows/PushWorkflow.yaml:50]: I think I missed this in the original workflow, but we should include the build reports for all of the modules. I think you can replace `app/build/reports` with `**/build/reports`.

**Review by temcguir — CHANGES_REQUESTED** [review]

**Review by temcguir — APPROVED** [review]

**Pull request merged**